### ISSUE TITLE
The i386 images are currently unbootable

### ISSUE THREAD

**Issue by probonopd** [furybsd/furybsd-livecd#184]
The i386 images are currently unbootable. Why?

**Comment by pkgdemon** [comment]
The script used to make a single hybrid optical/USB image is specific to amd64.  https://github.com/freebsd/freebsd/blob/master/release/amd64/mkisoimages.sh

To support other platforms including ARM it's likely we we will need to go back to the previous ways of making separate .iso / .img files.  Perhaps this script can be adapted for i386 but I am not sure offhand.

**Comment by pkgdemon** [comment]
It's also possible ARM using something entirely different I need to investigate it more to be honest.

**Comment by probonopd** [comment]
There is also https://github.com/freebsd/freebsd/blob/master/release/i386/mkisoimages.sh, maybe we need to use that for i386?

**Comment by pkgdemon** [comment]
Yep I would bet that is the issue.  Do you know if cirrius-ci bundles /usr/src?  Does it include support for non amd64 FreeBSD that we can just use the file in /usr/src/release/i386/mkisoimages.sh for example if it exists, and we can require that src.txz distribution be installed for builders?  Or do we need to look at shipping i386 version of the script under scripts as an alternative?

**Comment by probonopd** [comment]
Just switching to https://github.com/freebsd/freebsd/blob/master/release/i386/mkisoimages.sh does not make it boot on a non-EFI i386 machine, there must be something else going on.

Or does https://github.com/freebsd/freebsd/blob/master/release/i386/mkisoimages.sh only work if we burn to an actual CD-ROM, err, DVD?

**Comment by pkgdemon** [comment]
> Just switching to https://github.com/freebsd/freebsd/blob/master/release/i386/mkisoimages.sh does not make it boot on a non-EFI i386 machine, there must be something else going on.
> 
> Or does https://github.com/freebsd/freebsd/blob/master/release/i386/mkisoimages.sh only work if we burn to an actual CD-ROM, err, DVD?

I believe that is the case yes.  I want to say only AMD64 has hybrid ISO/USB image.  To make USB for i386 we have to run a separate script.  I would have to dig up those commands.

**Comment by probonopd** [comment]
Indeed. I was able to boot a i386 image from DVD-RW.

Hybrid ISO/USB images would be nice for i386.

**Comment by pkgdemon** [comment]
Pulling this conversation over from https://github.com/furybsd/furybsd-livecd/pull/187 for tracking this issue specifically.

As discussed in IRC different major versions of FreeBSD will have different versions of the mkisoimages.sh script. It adds another point of failure, maintenance.

https://github.com/freebsd/freebsd/blob/releng/12.1/release/amd64/mkisoimages.sh
https://github.com/freebsd/freebsd/blob/master/release/amd64/mkisoimages.sh

Also bringing in i386 support for hybrid ISO into FuryBSD itself will not be trivial as there were a number of commits in FreeBSD to add this support. The following may not be a complete list but it should give a rough idea.

https://reviews.freebsd.org/D15051
https://reviews.freebsd.org/rS332661
https://reviews.freebsd.org/rS331949
https://reviews.freebsd.org/D14799
https://reviews.freebsd.org/rS332084

Second I think as discussed we should using grub2 for at least the iso media. This would more simply address hybrid boot media needs for i386 as well as likely enhance compatibility issues where bsdloader fails. Or drop the idea of support i386, or pursue that fix properly in FreeBSD. Those are the options I think. For now we can take this work as is I just don't think it's optimal for proper i386 support, and may add a maintenance burden in the near future.
